### ISSUE TITLE
Delete alive data and late history

### ISSUE THREAD

**Issue by midenok** [tempesta-tech/mariadb#134]
Currently truncation works for some early points. There should be ability to truncate arbitrary portions of history:
```sql
truncate table ... with system versioning between ... and ...;
```
Possible variation may be:
```sql
truncate table ... with system versioning after ...;
```
F.ex. useful for archiving history. Second variant may be sufficient, should ask Alvin.

**Comment by midenok** [comment]
Related to #111, #82.

**Comment by krizhanovsky** [comment]
Originally in the HLD we had

    II.j. FLEXIBLE HISTORY PURGING
    ------------------------------
    
    A user should should be able to purge history on more flexible way, for
    example:
    
       > ALTER TABLE employee TRUNCATE SYSTEM VERSIONING WHERE TRX_ID < 20000 AND TRX_ID >= 10000;
       > ALTER TABLE employee TRUNCATE SYSTEM VERSIONING WHERE TIMESTAMP < '2016-02-06 10:03:03';
        
    alternatively, if this syntax is too strange, we can use syntax like this:

       > TRUNCATE TABLE employee WITH SYSTEM VERSIONING BETWEEN TRX_ID 10000 AND TRX_ID 20000;
       > TRUNCATE TABLE employee WITH SYSTEM VERSIONING BEFORE TIMESTAMP '2016-02-06 10:03:03';
    
    Such ALTER or TRUNCATE shouldn't be recorded in history itself.
    
    This will be implemented in step 0.80 of the development.

Thus, now we're going by the second way and BEFORE only is implemented. The questions is whether do we still need BETWEEN statement and do we need AFTER statement as opposite to BEFORE?

**Comment by kevgs** [comment]
Note on `BEFORE` implementation.

It deletes rows with `sys_trx_end < some_timestamp`. So after `TRUNCATE BEFORE '2017-1-1'` you still may see rows `AS OF TIMESTAMP '2017-1-1'`.
